apps/camera/js/controllers/settings.js

```diff
@@ -304,24 +304,24 @@ SettingsController.prototype.notify = function(setting, flashDeactivated) {
   if (dontNotify) { return; }
 
   var localizeOption = setting.get('optionsLocalizable') !== false;
-  var title = this.l10nGet(setting.get('title'));
-  var optionTitle = setting.selected('title');
-  var html;
-
+  var title = '<span data-l10n-id="' + setting.get('title') + '"></span>';
   // Localize option title only if not specified in the config
-  optionTitle = localizeOption ? this.l10nGet(optionTitle) : optionTitle;
+  var optionTitle = localizeOption ? 
+    '<span data-l10n-id="' + setting.selected('title') + '"></span>' :
+    '<span>' + optionTitle + '</span>';
+  var html;
 
   // Check if the `flashMode` setting is going to be deactivated as part
   // of the change in the `hdr` setting and display a specialized
   // notification if that is the case
   if (flashDeactivated) {
     html = title + ' ' + optionTitle + '<br/>' +
-      this.l10nGet('flash-deactivated');
+      '<span data-l10n-id="flash-deactivated"></span>';
   } else {
     html = title + '<br/>' + optionTitle;
   }
 
-  this.notification.display({ text: html });
+  this.notification.display({ text: {html: html} });
```

Comment:
the reason why I went for this is to match https://developer.mozilla.org/en-US/Firefox_OS/Developing_Gaia/localization_code_best_practices#Writing_APIs_that_operate_on_L10nIDs

We don't specify it as a separate class, because that would cost mem, but in general, we use a L10nID object type that can be a string - 'l10n-id' or an object that looks like this:
 - {id: 'l10n-id', args: {}}
 - {raw: 'raw string'}
 - {html: 'html string'}

We start by supporting only types that you need, but this enables the API to be predictable and future compatible if you later will need to support more of it.

What you are suggesting here will work for this case, but will be different from what we're exposing everywhere.

What I'm suggesting in this patch is making "text" element of the options object be of that l10nId object type.
This way, if you will have another l10nId type (subText for example) later, you just treat it as L10nID object type and cover all scenarios.

Does it make sense?